### PR TITLE
[MIG][10.0] base_partner_sequence

### PR DESCRIPTION
**Pull request by grindtildeath**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please take latest commits from 9.0 according migration guide: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0

**Comment by pedrobaeza** [comment]
I repeat: please take latest commits from 9.0, and you won't need to do most of the things you are doing.

File: base_partner_sequence/models/partner.py [hunk 1]
```diff
@@ -45,12 +45,10 @@ def copy(self, default=None):
 
     @api.multi
     def write(self, vals):
-        for partner in self:
-            ref = vals.get('ref') if 'ref' in vals else partner.ref
-            if not ref and self._needsRef(partner.id, vals):
-                vals['ref'] = self._get_next_ref(partner, vals)
-            super(ResPartner, partner).write(vals)
-        return True
+        if not vals.get('ref') and self._needsRef(vals):
```
Comment (pedrobaeza) [inline base_partner_sequence/models/partner.py]: Don't do this not multi-record aware.

**Comment by grindtildeath** [comment]
@pedrobaeza 
Thanks for your comments and sorry for the commit mess, but I'm new and unfortunately hadn't seen the port to v9.
Thanks to @guewen for his help.

**Review by guewen — APPROVED** [review]

**Review by astirpe — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
I have squashed 10.0 commits to a relevant one. Merging when CIs finish

**Pull request merged**